rails_programming/mailers_advanced_topics/websockets_and_actioncable.md

```diff
@@ -8,12 +8,12 @@ There are often times where you want to keep that connection open so that the se
 Look through these now and then use them to test yourself after doing the assignment.
 
 * What is a WebSocket?
+* What kinds of problems can WebSockets help you solve?
 * What options did developers have before WebSockets to update a client without a user request?
-* What is a consumer?
-* What is a subscriber?
-* What is a channel?
-* What is a stream?
 * How can you broadcast to a stream from the server?
+* Where do you authorize incoming connections?
+* What are ActionCable’s stream options?
+* What is the difference between `stream_from` and `stream_for`?
```

Comment:
I see that there's a point of confusion here, we don't want Knowledge Checks to be a repeat of the Learning Outcomes, as each serves a different purpose. Check out [this short wiki post ](https://github.com/TheOdinProject/curriculum/wiki/Learning-Outcomes-vs-Knowledge-Checks) that I wrote (and have now added to all of these issues) about the roles we want each to fill going forward.

The Learning Outcomes I've highlighted here feel like they're too "Knowledge Checky" to me and I think they would benefit from being higher level. I understand that some of them were already here, but there's no better time to improve them than now!

Reply:
Okay, I've made some changes here. Hopefully, this is closer to what yall are looking for. Please let me know what else may need to be done to get this up to par!

Thanks for your time!